src/lib/mediator.ts

```diff
@@ -0,0 +1,60 @@
+import { MediatorEventOptions } from "typings/mediator"
+import EventEmitter from "eventemitter3"
+
+export interface Mediator {
+  emitter: EventEmitter
+  trigger: (action: string, options?: MediatorEventOptions) => void
+  on: (event: string, cb?: (payload?: MediatorEventOptions) => void) => void
```

Comment:
We may want to. reconsider the callback type here and use `unknown` to guarantee that the person subscribing to the event verifies the response type before attempt to access potentially `undefined` values.

```ts
const payload: BuyerPremiumEventOptions = {
  auctionId: ''
}

trigger('test', payload)

on('test', (res: ArtworkEventOptions) => {
  console.log(res.artworkId)
})
```